pandas/core/indexes/base.py

```diff
@@ -5250,7 +5250,8 @@ def _raise_scalar_data_error(cls, data):
         #  in order to keep mypy happy
         raise TypeError(
             f"{cls.__name__}(...) must be called with a collection of some "
-            f"kind, {repr(data)} was passed"
+            f"kind, {repr(data) if not isinstance(data, np.generic) else str(data)} "
```

Comment:
does str work unconditionally?

Reply:
Yeah according to the linked PR seems so

Reply:
the can we simplify this?

Reply:
Oh sorry, str works unconditionally for numpy scalars but `repr` vs `str` would make this display differently for other scalars e.g. python strings